core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedScheduleGenerator.scala

```diff
@@ -606,8 +610,8 @@ class CostBasedScheduleGenerator(
       .map(level =>
         level
           .map(region => {
-            val (resourceConfig, regionCost) =
-              costEstimator.allocateResourcesAndEstimateCost(region, 1)
+            val (resourceConfig, regionCost) = costEstimatorMemoization
```

Comment:
When do we clean this cache?

Reply:
It has the same lifecycle as `CostBasedScheduleGenerator` and it needs to be used throughout the search process. As a result it is cleaned after the scheduler generates a schedule.